sdks/python/gen_protos.py

```diff
@@ -227,167 +291,270 @@ def _find_protoc_gen_mypy():
   for path in search_paths:
     fullpath = os.path.join(path, fname)
     if os.path.exists(fullpath):
+      LOG.info('Found protoc_gen_mypy at %s' % fullpath)
       return fullpath
-  raise RuntimeError("Could not find %s in %s" %
-                     (fname, ', '.join(search_paths)))
+  raise RuntimeError(
+      "Could not find %s in %s" % (fname, ', '.join(search_paths)))
+
 
+def find_by_ext(root_dir, ext):
+  for root, _, files in os.walk(root_dir):
+    for file in files:
+      if file.endswith(ext):
+        yield clean_path(os.path.join(root, file))
 
-def generate_proto_files(force=False, log=None):
 
+def ensure_grpcio_exists():
   try:
-    import grpc_tools  # pylint: disable=unused-import
+    from grpc_tools import protoc  # pylint: disable=unused-import
   except ImportError:
-    warnings.warn('Installing grpcio-tools is recommended for development.')
+    if platform.system() == 'Windows':
+      # For Windows, grpcio-tools has to be installed manually.
+      raise RuntimeError(
+          'Cannot generate protos for Windows since grpcio-tools package is '
+          'not installed. Please install this package manually '
+          'using \'pip install grpcio-tools\'.')
+    return _install_grpcio_tools()
 
-  if log is None:
-    logging.basicConfig()
-    log = logging.getLogger(__name__)
-    log.setLevel(logging.INFO)
+  return ""
```

Comment:
Returning the empty string seems odd here--best to return None or omit it. (It would be natural to let PythonPath handle None as a no-op.)

Reply:
fixed